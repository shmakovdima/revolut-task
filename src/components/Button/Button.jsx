import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Button = ({
  className, onClick, disabled, children,
}) => (
  <StyledButton
    disabled={disabled}
    onClick={!disabled ? onClick : null}
    className={className}
  >
    {children}
  </StyledButton>
);

Button.defaultProps = {
  disabled: false,
  className: null,
  onClick: () => {},
};

Button.propTypes = {
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

const StyledButton = styled.button`
  outline: none;
  border: none;
  padding: 16px 80px;
  color: white;
  font-weight: 500;
  font-size: 1rem;
  border-radius: ${props => props.theme.borderRadius};
  background: ${props => (props.disabled ? props.theme.disabledPink : props.theme.pink)};
  cursor: ${props => (props.disabled ? 'default' : 'pointer')};
  transition: all ${props => props.theme.transitionTime} ease 0s;

  &:hover {
    background: ${props => !props.disabled && props.theme.darkPink};
  }
`;

export default React.memo(Button);
