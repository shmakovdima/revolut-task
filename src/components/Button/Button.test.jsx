import React from 'react';
import { render, shallow, mount } from 'enzyme';

import Button from '.';

describe('Component [Button]', () => {
  test('Basic render', () => {
    const component = render(<Button>Test</Button>);
    expect(component).toMatchSnapshot();
  });

  test('Disabled prop', () => {
    const component = render(<Button disabled>Test</Button>);
    expect(component).toMatchSnapshot();
  });

  test('ClassName prop', () => {
    const className = 'test-class';
    const component = mount(<Button className={className}>ClassName</Button>);
    expect(component.hasClass(className)).toEqual(true);
    expect(component).toMatchSnapshot();
  });

  test('handleClick', () => {
    const handleClick = jest.fn();
    const component = shallow(<Button onClick={handleClick}>Test</Button>);
    component.simulate('click');
    expect(handleClick).toHaveBeenCalled();
  });

  test('handleClick disabled', () => {
    const handleClick = jest.fn();
    const component = shallow(
      <Button disabled onClick={handleClick}>
        Test
      </Button>,
    );
    component.simulate('click');
    expect(handleClick).not.toHaveBeenCalled();
  });
});
