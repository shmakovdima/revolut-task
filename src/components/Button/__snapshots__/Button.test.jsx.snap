// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component [Button] Basic render 1`] = `
<button
  class="sc-bdVaJa fYoLur"
>
  Test
</button>
`;

exports[`Component [Button] ClassName prop 1`] = `
<Button
  className="test-class"
>
  <styled.button
    className="test-class"
    disabled={false}
    onClick={[Function]}
  >
    <StyledComponent
      className="test-class"
      disabled={false}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "fYoLur",
            "rules": Array [
              "
  outline: none;
  border: none;
  padding: 16px 80px;
  color: white;
  font-weight: 500;
  font-size: 1rem;
  border-radius: ",
              [Function],
              ";
  background: ",
              [Function],
              ";
  cursor: ",
              [Function],
              ";
  transition: all ",
              [Function],
              " ease 0s;

  &:hover {
    background: ",
              [Function],
              ";
  }
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
    >
      <button
        className="test-class sc-bdVaJa fYoLur"
        disabled={false}
        onClick={[Function]}
      >
        ClassName
      </button>
    </StyledComponent>
  </styled.button>
</Button>
`;

exports[`Component [Button] Disabled prop 1`] = `
<button
  class="sc-bdVaJa dTmBxK"
  disabled=""
>
  Test
</button>
`;
